### ISSUE TITLE
reprex cannot generate a reprex of itself

### ISSUE THREAD

**Issue by dgrtwo** [jennybc/reprex#10]
```
reprex(reprex(1 + 1))
```

gives the error

```
 Error in env$metadata <- list() : 
    cannot change value of locked binding for 'metadata' 
```

(I wanted to reprex this example but I _couldn't!!_)

**Comment by jennybc** [comment]
Oh yes I am painfully familiar with the dreaded "nested render" problem. I even had to deal with it to create the README! Watch me [do gymnastics](https://github.com/jennybc/reprex/blob/master/README.Rmd).

See https://github.com/rstudio/rmarkdown/issues/248, for starters.

**Comment by jennybc** [comment]
This IS truly annoying. I wonder if we can develop the workaround into something in this package so that `reprex` itself is `reprex`able.

**Comment by jennybc** [comment]
OMG this works now. Here's a reprex of reprex making a reprex of a reprex.

If I execute this:

``` r
reprex(writeLines(reprex::reprex(1 + 1)))
```

this is displayed and put on the clipboard:

``` r
writeLines(reprex::reprex(1 + 1))
#> ``` r
#> 1 + 1
#> #> [1] 2
#> ```
```

Fixed via 5610d2401a6b65d408cf6dd01e671c7f1930ee21.
